lib/merchant_repository.rb

```diff
@@ -1,9 +1,55 @@
+require './lib/merchant'
+
 class MerchantRepository
-  attr_reader :csv_array,
-              :created_merchants
 
   def initialize(csv_array)
     @csv_array = csv_array
-    @created_merchants = []
+  end
+
+  def all
+    @csv_array
+  end
+
+  def find_by_id(id)
+    @csv_array.find do |merchant|
+      merchant.id == id
+    end
+  end
+
+  def find_by_name(name)
+    @csv_array.find do |merchant|
+      merchant.name.downcase == name.downcase
+    end
+  end
+
+  def find_all_by_name(name)
+    @csv_array.find_all do |merchant|
+      merchant.name.downcase.include?(name.downcase)
+    end
+  end
```

Comment:
I'm not saying we do this now, but maybe we could have name.downcase.include?(name.downcase) be a helper method inside of the Merchant class?